### PR TITLE
Plat 9889 : Add support in eSearch aggregations

### PR DESCRIPTION
**Pull request by MosheMaorKaltura**

### CODE REVIEW

File: plugins/search/providers/elastic_search/services/ESearchService.php [hunk 1]
```diff
@@ -70,7 +75,7 @@ function searchUserAction(KalturaESearchUserParams $searchParams, KalturaPager $
 	public function entryExportToCsvAction (KalturaMediaEsearchExportToCsvJobData $data)
 	{
 		if(!$data->userName || !$data->userMail)
-			throw new KalturaAPIException(APIErrors::USER_EMAIL_NOT_FOUND, $kuser);
+			throw new KalturaAPIException(APIErrors::USER_EMAIL_NOT_FOUND);
```
Comment (ZurKaltura) [inline plugins/search/providers/elastic_search/services/ESearchService.php]: USER_EMAIL_NOT_FOUND expect @USER_ID@ if we don't have $kuser maybe we should use different exception messages

File: plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+	}
+
+	public function coreToApiResponse($coreResponse, $fieldName = null)
+	{
+		$agg = new KalturaESearchAggregationResponseItem();
+		$agg->fieldName = $fieldName;
+		$agg->name = ESearchCategoryAggregationItem::KEY;
+		$bucketsArray = new KalturaESearchAggregationBucketsArray();
+		$buckets = $coreResponse[ESearchAggregations::BUCKETS];
```
Comment (gotlieb) [inline plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php]: $coreResponse should be validated.

File: plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		if ($buckets)
+		{
+			foreach ($buckets as $bucket)
+			{
+				$responseBucket = new KalturaESearchAggregationBucket();
+				$responseBucket->fromArray($bucket);
+				list(,$categoryName) = elasticSearchUtils::reverseFormatCategoryNameStatus($responseBucket->value);
```
Comment (gotlieb) [inline plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php]: what is the first parameter that is returned from the list? if not needed then why return 2 objects?

File: plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchEntryAggregationItem.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	public function coreToApiResponse($coreResponse, $fieldName = null)
+	{
+		$agg = new KalturaESearchAggregationResponseItem();
+		$agg->fieldName = $fieldName;
+		$agg->name = ESearchEntryAggregationItem::KEY;
+		$bucketsArray = new KalturaESearchAggregationBucketsArray();
+		$buckets = $coreResponse[ESearchAggregations::BUCKETS];
+		if ($buckets)
+		{
+
```
Comment (gotlieb) [inline plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchEntryAggregationItem.php]: remove line

File: plugins/search/providers/elastic_search/lib/model/aggregations/items/ESearchAggregationItem.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+{
+	const DEFAULT_SIZE = 5;
+	const NESTED_BUCKET = 'NestedBucket';
+
+	/**
+	 * @var int
+	 */
+	protected $size;
+
+	protected $fieldName;
```
Comment (gotlieb) [inline plugins/search/providers/elastic_search/lib/model/aggregations/items/ESearchAggregationItem.php]: missing @var type

File: plugins/search/providers/elastic_search/lib/model/aggregations/items/ESearchAggregationItem.php [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+	/**
+	 * @param int $size
+	 */
+	public function setSize($size)
+	{
+		$this->size = $size;
+	}
+
+	public function getFieldName()
```
Comment (gotlieb) [inline plugins/search/providers/elastic_search/lib/model/aggregations/items/ESearchAggregationItem.php]: missing notations

File: plugins/search/providers/elastic_search/lib/model/aggregations/items/ESearchAggregationItem.php [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+	{
+		return $this->fieldName;
+	}
+
+	public function setFieldName($fieldName)
```
Comment (gotlieb) [inline plugins/search/providers/elastic_search/lib/model/aggregations/items/ESearchAggregationItem.php]: missing notations

File: plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCuepointsAggregationItem.php [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+				$responseBucket->fromArray($bucket);
+				if($fieldName === ESearchCuePointsAggregationFieldName::TYPE)
+				{
+					$responseBucket->value =  kPluginableEnumsManager::coreToApi('CuePointType',$responseBucket->value);
+				}
+				$bucketsArray[] = $responseBucket;
+			}
+		}
+		$agg->buckets = $bucketsArray;
+		return ($agg);
```
Comment (MosheMaorKaltura) [inline plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCuepointsAggregationItem.php]: Check if array is needed here

File: plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCuepointsAggregationItem.php [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+		$bucketsArray = new KalturaESearchAggregationBucketsArray();
+		$buckets = $coreResponse[ESearchAggregationItem::NESTED_BUCKET][ESearchAggregations::BUCKETS];
+		if ($buckets)
+		{
+			foreach ($buckets as $bucket)
+			{
+				$responseBucket = new KalturaESearchAggregationBucket();
+				$responseBucket->fromArray($bucket);
+				if($fieldName === ESearchCuePointsAggregationFieldName::TYPE)
```
[the remark this answers is not in the document]
Comment (MosheMaorKaltura) [reply plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCuepointsAggregationItem.php]: Checking the returned value from the core.

File: plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php (hunk 3 above)
Comment (MosheMaorKaltura) [reply plugins/search/providers/elastic_search/lib/api/types/aggregations/items/KalturaESearchCategoryAggregationItem.php]: Future use

**Pull request merged**